swarm/pss/prox_test.go

```diff
@@ -133,99 +123,132 @@ func readSnapshot(t *testing.T, nodeCount int) simulations.Snapshot {
 	}
+	t.Run("8/100", testProxNetwork)
+	t.Run("16/100", testProxNetwork)
+	t.Run("32/100", testProxNetwork)
+	t.Run("64/100", testProxNetwork)
 }
 
-// This tests generates a sequenced number of messages with random addresses.
-// It then calculates which nodes in the network have the address of each message
-// within their nearest neighborhood depth, and stores them as recipients.
-// Upon sending the messages, it verifies that the respective message is passed to the message handlers of these recipients.
-// It will fail if a recipient handles a message it should not, or if after propagation not all expected messages are handled (timeout)
+// This tests generates a number of messages with random addresses. Then,
+// for each message it calculates which nodes in the network the msg address
+// within its nearest neighborhood depth, and stores those nodes as possible
+// recipients. Those nodes that are the closest to the message address (nodes
+// belonging to the deepest PO wrt msg address) are stored as required recipients.
+// Upon sending the messages, the test verifies that the respective message is
+// passed to the message handlers of these required recipients. Test will fail
+// if a message is handled by recipient which is not listed among the allowed
+// recipients of this particular message. It also fails after timeout, if not
+// all the required recipients have received their respective messages.
+//
+// For example, if proximity order of certain msg address is 4, and node X
+// has PO=5 wrt the message address, and nodes Y and Z have PO=6, then:
+// nodes Y and Z will be considered required recipients of the msg,
+// whereas nodes X, Y and Z will be allowed recipients.
 func testProxNetwork(t *testing.T) {
-	resetTestVariables()
+	var tstdata testData
 	msgCount, nodeCount := getCmdParams(t)
 	handlerContextFuncs := make(map[Topic]handlerContextFunc)
 	handlerContextFuncs[topic] = nodeMsgHandler
-	services := newProxServices(true, handlerContextFuncs, kademlias)
-	snap := readSnapshot(t, nodeCount)
-	sim = simulation.New(services)
-	defer sim.Close()
-	err := sim.Net.Load(&snap)
+	tstdata.kademlias = make(map[enode.ID]*network.Kademlia)
+	services := newProxServices(&tstdata, true, handlerContextFuncs, tstdata.kademlias)
+	tstdata.sim = simulation.New(services)
+	defer tstdata.sim.Close()
+	err := tstdata.sim.UploadSnapshot(fmt.Sprintf("testdata/snapshot_%d.json", nodeCount))
 	if err != nil {
 		t.Fatal(err)
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*60)
 	defer cancel()
-	err = sim.WaitTillSnapshotRecreated(ctx, snap)
+	snap := readSnapshot(t, nodeCount)
+	err = tstdata.sim.WaitTillSnapshotRecreated(ctx, snap)
 	if err != nil {
 		t.Fatalf("failed to recreate snapshot: %s", err)
 	}
-	assingTestVariables(sim, msgCount)
-	result := sim.Run(ctx, runFunc)
+	initializeTestData(&tstdata, msgCount)
```

Comment:
Again, please consider inline comments in the tests. Here, for example, 

```
// initialize and run the test
```

it may seem mundane, but I find it really helps readability.